e2e_samples/dataset_versioning/infra/README.md

```diff
@@ -0,0 +1,138 @@
+# Infrastructure as Code
+
+## Getting Started
+
+**Prerequisites:**
+
+- Terraform > 0.14.7
+- Azure cli > 2.19.1
+
+**Deploy Steps:**
+
+The following deploys the Azure base infrastructure for the Temperature Events sample. Ensure you are in `e2e_samples/dataset-versioning/infra/`.
+
+1. `az login`
+2. `./setup.sh`
+3. Run terraform init with the output from step 2
+4. `terraform validate && terraform plan && terraform apply`
+
+### #1 `az login`
+
+```bash
+> az login
+```
+
+Authenticate with the Azure CLI, to ensure you are able interact with your Azure account.
+<https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli>
+
+Before running the setup.sh script, make sure you are using the right subscription.
+You can check by using this command to get the list of subsciptions and see which one is the default:
+```az account list --output table```
+If you are using a different subscription and need to change it, use this command:
+```az account set --subscription <your_subscription_id>```
+
+### #2 `./setup.sh` to create Terraform support resources
+
+This script will create a resource group, with 2 resources (Key Vault and Storage) to track the secrets and state of your infrastructure. It will also create a new Service Principal with contributor permissions for the Terraform app to use later.
+
+```bash
+# Setup infrastructure for Terrafrom state & secrets.
+# NOTE: Change the values to a unique name for your account.
+# ./setup.sh 
+# Enter value: <resource group> <storage name> <Key Vault name> <region>
+$ az login
+$ ./setup.sh
+$ "Enter value: " rg-my-terraform stmyterraform kv-myterraform eastus2
+```
+
+Keep note of the output from `./setup.sh`. The output will be used in the next command
+
+### #3 Run terraform init with the output from step 2
+
+Use the output from the last command, to initialise the Terraform state files and store them in the Azure storage & Key Vault resources created in step 2.
+
+```bash
+# Note: This is an example, use the exact output given to you in step 2.
+az login
+cd terraform/live/dev
+
+terraform init -backend-config="storage_account_name=stmyterraform" -backend-config="container_name=terraform-state" -backend-config="access_key=$(az keyvault secret show --name tfstate-storage-key --vault-name kv-myterraform --query value -o tsv)" -backend-config="key=terraform.tfstate"
+```
+
+### #4 Use `terraform apply` to create the sample resources
+
+Run the following commands to deploy the sample infrastructure.
+**Note:** Pick a globally unique name and replace "your_rg_name" and "your_app_name"`
+
+```bash
+# NOTE: enter your own unique value instead of `myapp`
+terraform plan -var="rg_name=your_rg_name" -var="app_name=your_app_name" -out=/tmp/myapp
```

Comment:
Not a big deal, but due to the generated naming, this line does fail if the app_name is longer than 9 characters.